src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare = compareFunction || defaultCompareFunction;
+
+    function defaultCompareFunction(a, b) {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      } else if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
```

Comment:
The inner loop can be optimized to avoid unnecessary comparisons. Once the outer loop has completed an iteration, the last element is guaranteed to be in its correct position. Therefore, the inner loop should only iterate up to `this.length - i - 1` instead of `this.length - 1`.